使用加菜报错

使用"加菜 西北风"，机器人无回复后台报错。
07-30 18:59:59 [ERROR] nonebot | Running matcher <Matcher from nonebot_plugin_what2eat, type=message, priority=15, temp=False> failed.
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python3.9/multiprocessing/spawn.py", line 116, in spawn_main
    exitcode = _main(fd, parent_sentinel)
  File "/usr/lib/python3.9/multiprocessing/spawn.py", line 129, in _main
    return self._bootstrap(parent_sentinel)
  File "/usr/lib/python3.9/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3.9/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.9/dist-packages/uvicorn/subprocess.py", line 76, in subprocess_started
    target(sockets=sockets)
  File "/usr/local/lib/python3.9/dist-packages/uvicorn/server.py", line 60, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "/usr/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/local/lib/python3.9/dist-packages/nonebot/message.py", line 146, in _check_matcher
    await _run_matcher(Matcher, bot, event, state, stack, dependency_cache)
> File "/usr/local/lib/python3.9/dist-packages/nonebot/message.py", line 196, in _run_matcher
    await matcher.run(bot, event, state, stack, dependency_cache)
  File "/usr/local/lib/python3.9/dist-packages/nonebot/internal/matcher.py", line 668, in run
    await self.simple_run(bot, event, state, stack, dependency_cache)
  File "/usr/local/lib/python3.9/dist-packages/nonebot/internal/matcher.py", line 635, in simple_run
    await handler(
  File "/usr/local/lib/python3.9/dist-packages/nonebot/dependencies/__init__.py", line 92, in __call__
    return await self.call(**values)
  File "/usr/local/lib/python3.9/dist-packages/nonebot_plugin_what2eat/__init__.py", line 69, in _
    msg = eating_manager.add_basic_food(new_food)
  File "/usr/local/lib/python3.9/dist-packages/nonebot_plugin_what2eat/data_source.py", line 150, in add_basic_food
    status: FoodLoc = self._is_food_exists(new_food)
TypeError: _is_food_exists() missing 1 required positional argument: 'gid'

--- a/README.md
+++ b/README.md
@@ -20,7 +20,7 @@ _🧃🧋🍔🌮🍜🍮🍣🍻🍩 今天吃/喝什么 🍩🍻🍣🍮🍜�
     <img src="https://img.shields.io/badge/nonebot2-2.0.0b3+-green">
   </a>
   
-  <a href="https://github.com/MinatoAquaCrews/nonebot_plugin_what2eat/releases/tag/v0.3.3rc2">
+  <a href="https://github.com/MinatoAquaCrews/nonebot_plugin_what2eat/releases/tag/v0.3.3">
     <img src="https://img.shields.io/github/v/release/MinatoAquaCrews/nonebot_plugin_what2eat?color=orange">
   </a>
 
@@ -32,11 +32,11 @@ _🧃🧋🍔🌮🍜🍮🍣🍻🍩 今天吃/喝什么 🍩🍻🍣🍮🍜�
 
 ## 版本
 
-v0.3.3rc2 今天喝什么！
+v0.3.3 今天喝什么！
 
 ⚠ 适配nonebot2-2.0.0b3+
 
-[更新日志](https://github.com/MinatoAquaCrews/nonebot_plugin_what2eat/releases/tag/v0.3.3rc2)
+[更新日志](https://github.com/MinatoAquaCrews/nonebot_plugin_what2eat/releases/tag/v0.3.3)
 
 ## 安装
 
@@ -63,6 +63,7 @@ v0.3.3rc2 今天喝什么！
     EATING_LIMIT=5									# 每个时段吃什么的次数上限，默认5次；每日6点、11点、17点、22点自动刷新
     GREETING_GROUPS_ID=["123456789", "987654321"]	# 默认开启小助手群组，或{"123456789", "987654321"}
     SUPERUSERS={"12345678"}                       	# 同nonebot超管配置
+    NICKNAME={"Bot"}                                # 同nonebot机器人昵称配置
     ```
 
 3. 群管理可自行添加或移除群特色菜单（位于`eating.json`下`[group_food][group_id]`）；超管可添加或移除基础菜单（`[basic_food]`）；
@@ -77,18 +78,16 @@ v0.3.3rc2 今天喝什么！
 
     ⚠ 群吃饭小助手启用配置存于`greetings.json`的`groups_id`字段
 
-6. 初次使用该插件时，若不存在`eating.json`与`greetings.json`文件，设置`USE_PRESET_MENU`及`USE_PRESET_GREETINGS`可从仓库中尝试下载；若下载失败则会报错。会尝试从仓库中下载`drinks.json`。
+6. 初次使用该插件时，若不存在`eating.json`与`greetings.json`文件，设置`USE_PRESET_MENU`及`USE_PRESET_GREETINGS`可从仓库中尝试下载；会尝试从仓库中下载`drinks.json`。若资源下载失败且本地不存在，则抛出错误。
 
     ```python
     USE_PRESET_MENU=false
     USE_PRESET_GREETINGS=false
     ```
 
-    ⚠ 若存在`eating.json`与`greetings.json`，不会从仓库中下载，仅检查是否缺少必要的键值，并补写初始键值
-
     ⚠ 从仓库下载会**覆写**原有配置！建议老用户按需开启从仓库下载设置
 
-    ⚠ `v0.3.3`已弃用兼容，请注意
+    ⚠ `v0.3.3`已弃用`v0.2.x`版本旧资源的兼容，请注意
 
 ## 命令
 
@@ -96,19 +95,19 @@ v0.3.3rc2 今天喝什么！
 
 2. 🔥 **新增** 喝什么： 今天喝什么、中午喝啥、今晚喝啥、中午喝什么、晚上喝啥、晚上喝什么、夜宵喝啥……
 
-    ⚠ 喝什么与吃什么共用`EATING_LIMIT`次数
+    ⚠ 与吃什么共用`EATING_LIMIT`次数
 
 3. [管理员或超管] 添加或移除群菜名：[添加/移除 菜名]；
 
 4. 查看群菜单：[菜单/群菜单/查看菜单]；
 
-6. [超管] 添加至基础菜单：[加菜 菜名]；
+5. [超管] 添加至基础菜单：[加菜 菜名]；
 
-7. 查看基础菜单：[基础菜单]；
+6. 查看基础菜单：[基础菜单]；
 
-8. [管理员或超管] 开启/关闭吃饭小助手：[开启/启用/关闭/禁用小助手]；
+7. [管理员或超管] 开启/关闭吃饭小助手：[开启/启用/关闭/禁用小助手]；
 
-9. [管理员或超管] 添加/删除吃饭小助手问候语：[添加/删除/移除问候 时段 问候语]；
+8. [管理员或超管] 添加/删除吃饭小助手问候语：[添加/删除/移除问候 时段 问候语]；
 
     ⚠ 添加/移除问候操作可一步步进行，或一次性输入两或三个命令；可中途取消操作
 
@@ -120,7 +119,9 @@ v0.3.3rc2 今天喝什么！
 
     A：建议肯德基
 
-    （吃什么*5）
+    Q：今晚喝啥
+
+    A：不如来杯 茶颜悦色 的 幽兰拿铁 吧！
 
     Q：今晚吃什么
 

--- a/nonebot_plugin_what2eat/__init__.py
+++ b/nonebot_plugin_what2eat/__init__.py
@@ -1,15 +1,16 @@
-from typing import Coroutine, Any
+from typing import Coroutine, Any, List, Dict, Union
 from nonebot import on_command, on_regex, logger
 from nonebot.typing import T_State
 from nonebot.permission import SUPERUSER
-from nonebot.adapters.onebot.v11 import GROUP, GROUP_ADMIN, GROUP_OWNER, Message, MessageEvent, GroupMessageEvent
+from nonebot.adapters.onebot.v11 import Bot, GROUP, GROUP_ADMIN, GROUP_OWNER, Message, MessageEvent, MessageSegment, GroupMessageEvent
 from nonebot.params import Depends, Arg, ArgStr, CommandArg, RegexMatched
 from nonebot.matcher import Matcher
 from nonebot_plugin_apscheduler import scheduler
 from .utils import Meals
+from .config import what2eat_config
 from .data_source import eating_manager
 
-__what2eat_version__ = "v0.3.3rc2"
+__what2eat_version__ = "v0.3.3"
 __what2eat_notes__ = f'''
 今天吃什么？ {__what2eat_version__}
 [xx吃xx]    问bot吃什么
@@ -17,8 +18,8 @@ __what2eat_notes__ = f'''
 [添加 xx]   添加菜品至群菜单
 [移除 xx]   从菜单移除菜品
 [加菜 xx]   添加菜品至基础菜单
-[菜单]       查看群菜单
-[基础菜单]  查看基础菜单
+[菜单]        查看群菜单
+[基础菜单] 查看基础菜单
 [开启/关闭小助手] 开启/关闭吃饭小助手
 [添加/删除问候 时段 问候语] 添加/删除吃饭小助手问候语'''.strip()
 
@@ -94,15 +95,23 @@ async def _(event: GroupMessageEvent, args: Message = CommandArg()):
     await group_remove.finish(msg)
 
 @show_group_menu.handle()
-async def _(event: GroupMessageEvent):
+async def _(bot: Bot, event: GroupMessageEvent):
     gid = str(event.group_id)
-    msg = eating_manager.show_group_menu(gid)
-    await show_group_menu.finish(msg)
+    line, msg = eating_manager.show_group_menu(gid)
+    if line > 20:
+        chain = await chain_reply(bot, [], msg)
+        await bot.call_api("send_group_forward_msg", group_id=event.group_id, messages=chain)
+    else:
+        await show_group_menu.finish(msg)
 
 @show_basic_menu.handle()
-async def _(matcher: Matcher):
-    msg = eating_manager.show_basic_menu()
-    await matcher.finish(msg)
+async def _(bot: Bot, event: GroupMessageEvent):
+    line, msg = eating_manager.show_basic_menu()
+    if line > 20:
+        chain = await chain_reply(bot, [], msg)
+        await bot.call_api("send_group_forward_msg", group_id=event.group_id, messages=chain)
+    else:
+        await show_basic_menu.finish(msg)
 
 # ------------------------- Greetings -------------------------
 @greeting_on.handle()
@@ -254,3 +263,15 @@ async def time_for_dinner():
 async def time_for_midnight():
     await eating_manager.do_greeting(Meals.MIDNIGHT)
     logger.info(f"已群发夜宵提醒")
+
+async def chain_reply(bot: Bot, chain: List[Dict[str, Union[str, Dict[str, Union[str, MessageSegment]]]]], msg: MessageSegment) -> List[Dict[str, Union[str, Dict[str, Union[str, MessageSegment]]]]]:
+    data = {
+        "type": "node",
+        "data": {
+            "name": f"{list(what2eat_config.nickname)[0]}",
+            "uin": f"{bot.self_id}",
+            "content": msg
+        },
+    }
+    chain.append(data)
+    return chain

--- a/nonebot_plugin_what2eat/config.py
+++ b/nonebot_plugin_what2eat/config.py
@@ -16,6 +16,7 @@ class PluginConfig(BaseModel, extra=Extra.ignore):
     eating_limit: int = 5
     greeting_groups_id: Set[str] = set()
     superusers: Set[str] = set()
+    nickname: Set[str] = {"Bot"}
     
 driver = get_driver()
 what2eat_config: PluginConfig = PluginConfig.parse_obj(driver.config.dict())

--- a/nonebot_plugin_what2eat/data_source.py
+++ b/nonebot_plugin_what2eat/data_source.py
@@ -95,11 +95,11 @@ class EatingManager:
             save_json(self._eating_json, self._eating)
             return MessageSegment.text(random.choice(
                     [
-                        "你今天已经吃得够多了！",
-                        "吃这么多的吗？",
-                        "害搁这吃呢？不工作的吗？",
-                        "再吃肚子就要爆炸咯~",
-                        "你是米虫吗？今天碳水要爆炸啦！"
+                        "你今天已经喝得够多了！",
+                        "喝这么多的吗？",
+                        "害搁这喝呢？不工作的吗？",
+                        "再喝肚子就要爆炸咯~",
+                        "你是水桶吗？今天糖分要超标啦！"
                     ]
                 )
             )
@@ -111,7 +111,7 @@ class EatingManager:
             return MessageSegment.text(random.choice(
                     [
                         f"不如来杯 {_branch} 的 {_drink} 吧！",
-                        f"去 {_branch}整杯 {_drink} 吧！",
+                        f"去 {_branch} 整杯 {_drink} 吧！",
                         f"{_branch} 的 {_drink} 如何？"
                     ]
                 )
@@ -220,7 +220,7 @@ class EatingManager:
         return _branch, _drink
 
     # ------------------------- Menu -------------------------
-    def show_group_menu(self, gid: str) -> MessageSegment:
+    def show_group_menu(self, gid: str) -> Tuple[int, MessageSegment]:
         msg: str = ""
         self._eating = load_json(self._eating_json)
         self._init_data(gid)
@@ -231,11 +231,11 @@ class EatingManager:
             for food in self._eating["group_food"][gid]:
                 msg += f"\n{food}"
             
-            return MessageSegment.text(msg)
+            return len(self._eating["group_food"][gid]), MessageSegment.text(msg)
         
-        return MessageSegment.text("还没有群特色菜单呢，请[添加 菜名]🤤")
+        return 0, MessageSegment.text("还没有群特色菜单呢，请[添加 菜名]🤤")
 
-    def show_basic_menu(self) -> MessageSegment:
+    def show_basic_menu(self) -> Tuple[int, MessageSegment]:
         msg: str = ""
         self._eating = load_json(self._eating_json)
 
@@ -244,9 +244,9 @@ class EatingManager:
             for food in self._eating["basic_food"]:
                 msg += f"\n{food}"
             
-            return MessageSegment.text(msg)
+            return len(self._eating["basic_food"]), MessageSegment.text(msg)
         
-        return MessageSegment.text("还没有基础菜单呢，请[添加 菜名]🤤")
+        return 0, MessageSegment.text("还没有基础菜单呢，请[添加 菜名]🤤")
 
     # ------------------------- Greetings -------------------------
     def update_groups_on(self, gid: str, new_state: bool) -> None:
@@ -327,7 +327,7 @@ class EatingManager:
             for gid in self._greetings["groups_id"]:
                 if self._greetings["groups_id"].get(gid, False):
                     try:
-                        await bot.send_group_msg(group_id=int(gid), message=msg)
+                        await bot.call_api("send_group_msg", group_id=int(gid), message=msg)
                     except ActionFailed as e:
                         logger.warning(f"发送群 {gid} 失败：{e}")
     

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,7 +1,7 @@
 [tool.poetry]
 name = "nonebot_plugin_what2eat"
-version = "0.3.3rc2"
-description = "Ask bot for what to eat today!"
+version = "0.3.3"
+description = "Ask bot for what to eat or what to drink!"
 authors = ["KafCoppelia <k740677208@gmail.com>"]
 license = "MIT"
 readme = "README.md"

